lib/rubocop/cop/rspec_rails/inferred_spec_type.rb

```diff
@@ -1,143 +0,0 @@
-# frozen_string_literal: true
-
-module RuboCop
-  module Cop
-    module RSpecRails
-      # Identifies redundant spec type.
-      #
-      # After setting up rspec-rails, you will have enabled
-      # `config.infer_spec_type_from_file_location!` by default in
-      # spec/rails_helper.rb. This cop works in conjunction with this config.
-      # If you disable this config, disable this cop as well.
```

Comment:
The setting’s default has just been changed in a rspec-rails version.
There are, and in the next years there will be plenty of projects with rails_helper that has this option turned on.

And the option is still [there](https://github.com/rspec/rspec-rails/pull/2804/files#diff-86c6eceb5f674cb6e3590f6cf646ebaaf395fa52a578310c21e2af16814e75aeR85) to opt-in.

The cop is marked as unsafe, meaning it doesn’t raise offences normally.

Basing on all this, I suggest changing the documentation to reflect the above, keep the cop, keep the cop enabled.

Reply:
Here's the problem as I see it: the templates for generating specs (from scaffolding or just generating individual controllers, models, etc.) have the `type:` specifications, and they have had them for years. Keeping this cop enabled by default makes it seem like the best practice is to remove those `type` annotations from the spec. I know this because I have been doing it on a recent project. However, that isn't the recommended best practice (and hasn't been since before the cop's creation).

So if we keep the cop enabled by default, the worst thing that will happen is that people will get a warning and remove the type specifications believing that Rubocop's defaults are recommending best practices. This will be for both older and brand new projects since this cop doesn't check to see if `config.infer_spec_type_from_file_location` is set. 

If we remove the cop, the worst thing that will happen is that people who have `config.infer_spec_type_from_file_location` turned on won't get warned if they add a spec with a type specification.

Not being warned that a new spec is following best practices and is future-proofed against a deprecated option doesn't seem as bad as actively recommending that people use a long-deprecated feature.

Reply:
Fair enough.
A recent rspec-rails issue to support this https://github.com/rspec/rspec-rails/issues/2815#issuecomment-2496000959